build-tools/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -143,6 +143,7 @@
     private final LazyPropertyMap<String, File> extraConfigFiles = new LazyPropertyMap<>("Extra config files", this, FileEntry::new);
     private final LazyPropertyList<FileCollection> extraJarConfigurations = new LazyPropertyList<>("Extra jar files", this);
     private final List<Map<String, String>> credentials = new ArrayList<>();
+    private final List<File> roleFiles = new ArrayList<>();
```

Comment:
We need to expose this new configuration as Gradle input so that caching and build avoidance works properly. That ensures that changes to these files (or their removal) will trigger Gradle to run the tests again, or invalidate any cached results. Basically, we just need to add an appropriately annotated getter which exposes this.

```java
@InputFiles
@PathSensitive(PathSensitivity.RELATIVE)
public List<File> getRoleFiles() {
    return roleFiles;
}
```